### PR TITLE
Improve docs for contributing and testing

### PR DESCRIPTION
**Pull request by tkell**
Hello @rabitt and @drubinstein and @andreasjansson and friends!  This in-progress PR is going to take a shot at #44:

- [x] Break out CONTRIBUTING.md
- [x] Add info on how to run tests locally
- [x] Add info as per @lostanlen's request on what sort of functionality to test for when adding a new dataset.

Let me know if you think this is a good / bad idea!

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+Once pyenv and the three versions of Python are installed, install [tox](https://tox.readthedocs.io/en/latest/), our test runner.
+
+Finally, run tox with `tox`.  All tests should pass!
+
+
+## Contributing a new dataset loader
+
+To add a new dataset loader:
+1. Create an index in `mirdata/indexes`, e.g. `my_dataset_index.json`
```
Comment (rabitt) [inline CONTRIBUTING.md]: 1. Create a script in `scripts`, e.g. `make_my_dataset_index.py`, which generates an index file (see below)
2. Run the script on the canonical* version of the dataset and save the index in `mirdata/indexes` e.g. `my_dataset_index.json`.
3. Create a module in `mirdata`, e.g. `my_dataset.py`

*Whenever possible, this should be the official release of the dataset as published by the dataset creators. When this is not possible, (e.g. for data that is no longer available), find as many versions of the data as you can from different researchers (at least 4), and use the most common one.

Comment (rabitt) [reply CONTRIBUTING.md]: ^^ @keunwoochoi @magdalenafuentes controversial stuff in the *comment. What do you think?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+on string parsing on load.
+
+Scripts used to create the dataset indexes are in the [scripts](https://github.com/mir-dataset-loaders/mirdata/tree/master/scripts) folder.
+
+
+### Creating a module.
+
+Copy and paste this template and fill it in for your dataset:
+
+```python
```
Comment (rabitt) [inline CONTRIBUTING.md]: Womp womp...this has changed... 

I'll help rewrite this when I finish my current PR

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+TODO: The Bibtex format citation here
+"""
+
+    print(cite_data)
+```
+
+
+## Adding tests to your dataset
+
+Add some words about adding tests here!
```
Comment (rabitt) [inline CONTRIBUTING.md]: @tkell take a look at my tests for `ikala`. I think most people will be a bit overwhelmed by the pytest boilerplate, the mocking, etc. I think we should advise people to

1) make a fake/tiny version of the dataset in the tests folder `tests/resources/mir_datasets/my_dataset/`
2) test all of the dataset specific code, e.g. the Track object, any of the _load functions, etc, as I did in ikala.
3) let's give people permission to skip writing tests for the `download` and `validate` functions...maybe we can find a clever way to test all loader's `download` and `validate` functions in a separate test file

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
+# Contributing Code & Datasets
+
+## Installing and running tests
+
+First, check out the repository from Github.
```
[the remark this answers is not in the document]
Comment (tkell) [reply CONTRIBUTING.md]: Can do!

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+        track1.wav
+        track2.wav
+        track3.wav
+    > annotations/
+        track1.csv
+        Track2.csv
+        track3.csv
+```
+The top level directory is `Example_Dataset` and the relative path for `track1.wav`
+should be `Example_Dataset/audio/track1.wav`.
```
[the remark this answers is not in the document]
Comment (tkell) [reply CONTRIBUTING.md]: 🚀

File: CONTRIBUTING.md (hunk 1 above)
Comment (tkell) [reply CONTRIBUTING.md]: 🚀

File: CONTRIBUTING.md (hunk 2 above)
Comment (tkell) [reply CONTRIBUTING.md]: 🚀

File: CONTRIBUTING.md (hunk 3 above)
Comment (tkell) [reply CONTRIBUTING.md]: Took a shot at the copy for this - please correct the prose as needed.

**Comment by tkell** [comment]
As per an in-person chat, we will update CONTRIBUTING.md after @rabitt has landed #90, and then merge this.

File: CONTRIBUTING.md (hunk 2 above)
Comment (tkell) [reply CONTRIBUTING.md]: @rabitt, could I trouble you for some updated copy here?

File: CONTRIBUTING.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (tkell) [reply CONTRIBUTING.md]: @rabitt, could I trouble you for some updated copy here?

**Pull request merged**